lib/pdcleaner/pdcleaner.go

```diff
@@ -218,33 +221,89 @@ func (p *pdcleaner) CleanOnce() error {
 		if err != nil {
 			return fmt.Errorf("getting active sector sets for miner %s: %w", p.miner, err)
 		}
+		unProvenSectors, err := miner.AllPartSectors(mas, miner.Partition.UnprovenSectors)
+		if err != nil {
+			return fmt.Errorf("getting unproven sector sets for miner %s: %w", p.miner, err)
+		}
+		finalSectors, err := bitfield.MergeBitFields(activeSectors, unProvenSectors)
+		if err != nil {
+			return fmt.Errorf("merging bitfields to generate all deal sectors on miner %s: %w", p.miner, err)
+		}
+
+		// Load verifreg actor locally
+		verifregActor, err := p.full.StateGetActor(p.ctx, verifreg.Address, tskey)
+		if err != nil {
+			return fmt.Errorf("getting verified registry actor state: %w", err)
+		}
+		verifregState, err := verifreg.Load(store, verifregActor)
+		if err != nil {
+			return fmt.Errorf("loading verified registry actor state: %w", err)
+		}
 
 		for _, d := range completeDirectDeals {
-			// AllocationID and ClaimID should match
 			cID := verifregtypes.ClaimId(d.AllocationID)
 			c, ok := claims[cID]
+			// If claim found
 			if ok {
-				present, err := activeSectors.IsSet(uint64(c.Sector))
+				// Claim Sector number and Deal Sector number should match(regardless of how DDO works)
+				// If they don't match and older sector is removed, then we can't use the metadata
+				// This check can be removed once Curio has resealing enabled, and it can provide
```

Comment:
🤞 that you remember this! it'll be annoying if that lands and this is forgotten

Reply:
It is not just this but the normal deal flow as well. Everything will be impacted by Curio's robustness.